internal/elasticsearch/deployment.go

```diff
@@ -229,89 +224,61 @@ func (node *deploymentNode) unpause() error {
 }
 
 func (node *deploymentNode) setPaused(paused bool) error {
+	equalFunc := func(current, _ *apps.Deployment) bool {
+		return current.Spec.Paused == paused
+	}
+	mutateFunc := func(current, _ *apps.Deployment) {
+		current.Spec.Paused = paused
+	}
 	// we use pauseNode so that we don't revert any new changes that should be made and
 	// noticed in state()
-	pauseNode := node.self.DeepCopy()
-	ll := log.WithValues("node", pauseNode.Name)
-
-	nretries := -1
-	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		nretries++
-		if err := node.client.Get(context.TODO(), types.NamespacedName{Name: pauseNode.Name, Namespace: pauseNode.Namespace}, pauseNode); err != nil {
-			ll.Info("Could not get Elasticsearch node resource",
-				"error", err)
-			return err
-		}
-
-		if pauseNode.Spec.Paused == paused {
-			return nil
-		}
-
-		pauseNode.Spec.Paused = paused
-
-		if err := node.client.Update(context.TODO(), pauseNode); err != nil {
-			ll.Info("failed to update node resource",
-				"error", err)
-			return err
-		}
-		return nil
-	})
-	if retryErr != nil {
-		return kverrors.Wrap(retryErr, "could not update Elasticsearch node after retries",
-			"node", node.self.Name,
-			"retries", nretries,
+	pausedNode := node.self.DeepCopy()
+	err := deployment.Update(context.TODO(), node.client, pausedNode, equalFunc, mutateFunc)
+	if err != nil {
+		return kverrors.Wrap(err, "failed to update elasticsearch node deployment",
+			"cluster", node.clusterName,
+			"namespace", node.self.Namespace,
 		)
 	}
 
-	node.self.Spec.Paused = pauseNode.Spec.Paused
+	node.self.Spec.Paused = paused
```

Comment:
why do we have this if we also have the update function setting this? shouldn't we set the `node.self.Spec.Paused` to match what the paused node is?

Reply:
This is there for a long time now. This is passing the same `paused` value to `pausedNode.Spec.Paused` as well as to `node.self.Spec.Paused` if we successfully update the first one.

Reply:
Which update function do you mention here? `executeUpdate()`, `state()`?